apidef/migration.go

```diff
@@ -78,3 +78,37 @@ func (a *APIDefinition) MigrateVersioning() (versions []APIDefinition, err error
 
 	return
 }
+
+func (a *APIDefinition) MigrateCachePlugin() (err error) {
+	if a.VersionDefinition.Enabled || len(a.VersionDefinition.Versions) != 0 {
+		return errors.New("not migratable - new versioning is enabled")
+	}
+
+	if a.VersionData.NotVersioned && len(a.VersionData.Versions) > 1 {
+		return errors.New("not migratable - if not versioned, there should be 1 version info in versions map")
+	}
+
+	for vName, vInfo := range a.VersionData.Versions {
+		var updated = false
+		if vInfo.UseExtendedPaths && len(vInfo.ExtendedPaths.Cached) > 0 {
+			var methods = []CacheMeta{}
+			for _, cache := range vInfo.ExtendedPaths.Cached {
+				newCache := CacheMeta{
+					Path:     cache,
```

Comment:
I know in old cache there is no `method` but how will we write it in OAS format if method is empty? We need to find a solution here.

Reply:
As cache works only for the safe methods, `GET`,`HEAD`, `OPTIONS`. We need to add 3 advance cache for all.

https://github.com/TykTechnologies/tyk/blob/f603dec31b91ae9c7221c990b078d82aa04bfece/gateway/mw_redis_cache.go#L377-L379

Reply:
from where we can get info about method?

Reply:
As I said above, you don’t get info about method. We need to write it to 3 safe methods.